bugsnag-spring/src/test/java/com/bugsnag/NotifierTest.java

```diff
@@ -0,0 +1,49 @@
+package com.bugsnag;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.Collections;
+
+public class NotifierTest {
+
+    private Report report;
+    private Configuration config;
+    private SessionPayload sessionPayload;
+
+    /**
+     * Initialises the objectmapper + report for conversion to json
+     *
+     * @throws Throwable the throwable
+     */
+    @Before
+    public void setUp() throws Throwable {
+        config = new Configuration("api-key");
+        report = new Report(config, new RuntimeException());
+        sessionPayload = new SessionPayload(Collections.<SessionCount>emptyList(), config);
+    }
+
+    @Test
+    public void testNotificationSerialisation() throws Throwable {
+        JsonNode payload = BugsnagTestUtils.mapReportToJson(config, this.report);
+        JsonNode notifier = payload.get("notifier");
+
+        assertEquals("Bugsnag Spring", notifier.get("name").asText());
+        assertEquals("https://github.com/bugsnag/bugsnag-java", notifier.get("url").asText());
+        assertNotNull(notifier.get("version").asText());
```

Comment:
I would also check that the version is non-empty, same for sessions and the other NotifierTest class.